ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/Attachment.kt

```diff
@@ -0,0 +1,65 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.security.PublicKey
+import java.util.jar.JarInputStream
+
+/**
+ * Defines a data attachment that can be referenced within a transaction.
+ *
+ * Attachments are ZIP, or optionally a signed JAR file that contains one or more files of public, static data that
+ * can be referenced from a transaction, and utilised within a contract. Examples of attachments include:
+ *
+ * - Calendar data
+ * - Fixes (e.g. LIBOR)
+ * - Smart contract code
+ * - Legal documents
+ * - Facts generated by oracles.
+ *
+ * @property id The unique hash of the attachment.
+ * @property size The size of the attachment in bytes.
+ * @property signatories The keys that have been used to sign the attachment, or empty if the attachment does not require signing.
+ */
+@CordaSerializable
+@DoNotImplement
+interface Attachment {
+
+    val id: SecureHash
+    val size: Int
+    val signatories: Set<PublicKey>
+
+    /**
+     * Finds the specified file (case insensitively) within the attachment and copies it to the specified output stream.
+     *
+     * @param path The path of the file to find within the attachment.
+     * @param output The output stream where the file should be copied to.
+     * @throws [FileNotFoundException] if the specified path does not exist in the attachment.
+     */
+    fun extractFile(path: String, output: OutputStream)
```

Comment:
I think the interface should not limit the implementation to files.
It will help with future enhancement and mocking of tests ;)

Reply:
Attachments are zips/jars to they are necessarily file based, though I agree this should be URL rather than path